src/macaron/repo_finder/repo_validator.py

```diff
@@ -13,6 +13,8 @@
 def find_valid_repository_url(urls: Iterable[str]) -> str:
```

Comment:
I think it's important to make sure that the order of URLS obtained from the tags described in `repo_pom_paths` is kept when provided to `find_valid_repository_url`. Perhaps we need to add necessary tests to ensure that this behavior is maintained (unless there are existing tests for this behavior that I didn't remember :thinking: ).